scipy/stats/tests/test_stats.py

```diff
@@ -7917,6 +7917,222 @@ def test_alias(self):
         assert_equal(res.stat, res.statistic)
 
 
+class TestQuantileTest():
+    r""" Test the non-parametric quantile test, 
+    including the computation of confidence intervals
+    """
+    
+    np.random.seed(0)
+    m = 100                                     # number of tests
+    N = np.random.randint(10, 100, size=m)      # num of samples
+    x = []                                      # sorted samples
+    for i in range(m):
+        num = N[i]
+        x.append(np.sort(np.random.permutation(1000)[:num]))
+    quantile_def = np.random.rand(m)            # quantile definition in [0,1)
+    quantile_val = np.random.rand(m)*100        # quantile value in [0,100)
+    confidence = 0.25*np.random.rand(m)+0.75    # confidence in [0.75,1)
+
+    def test_quantile_test_iv(self):
+        
+        x = [1,2,3]
+        p = 0.5
+        q = 0
+        alternative = 'less'
+
+        message = "`x` must be a one-dimensional array of numbers."
+        with pytest.raises(ValueError, match=message):
+            stats.quantile_test([[1,2,3]],q,p,alternative=alternative)
+        
+        message = "`q` must be a scalar."
+        with pytest.raises(ValueError, match=message):
+            stats.quantile_test(x,[1,2],p,alternative=alternative)
+        
+        message = "`p` must be a float strictly between 0 and 1."
+        with pytest.raises(ValueError, match=message):
+            stats.quantile_test(x,q,[0.5,0.75],alternative=alternative)
+        with pytest.raises(ValueError, match=message):
+            stats.quantile_test(x,q,2,alternative=alternative)
+        with pytest.raises(ValueError, match=message):
+            stats.quantile_test(x,q,-2,alternative=alternative)
+        
+        message = "alternative not recognized; \n"
```

Comment:
```suggestion
        message = "alternative not recognized"
```
It's treated as a regular expression, so it doesn't need to be complete